html/captive-portal/templates/wireless-profile.xml

```diff
@@ -6,55 +6,145 @@
-                                <false/>
-                                <key>TLSAllowTrustExceptions</key>
-                                <true/>
+                                [% IF for_windows %]
                                 <key>UserName</key>
                                 <string>[% username %]</string>
-                                <key>UserPassword</key>
-                                <string></string>
+                                [% END %]
                         </dict>
+                        [% END %]
                         <key>EncryptionType</key>
-                        <string>WPA</string>
+                        <string>[% provisioner.security_type %]</string>
                         <key>HIDDEN_NETWORK</key>
-                        <true/>
+                        <false/>
                         <key>PayloadDescription</key>
                         <string>Configures wireless connectivity settings.</string>
                         <key>PayloadDisplayName</key>
                         <string>Wi-Fi ([% provisioner.ssid %])</string>
                         <key>PayloadIdentifier</key>
-                        <string>org.packetfence.profile.</string>
+                        <string>[% provisioner.reversedns %]</string>
                         <key>PayloadOrganization</key>
-                        <string></string>
+                        <string>[% provisioner.company %]</string>
                         <key>PayloadType</key>
                         <string>com.apple.wifi.managed</string>
                         <key>PayloadUUID</key>
                         <string>247F8812-0419-4548-9389-EA76325DEFC8</string>
                         <key>PayloadVersion</key>
                         <integer>1</integer>
+                        <key>PayloadCertificateUUID</key>
+                        <string>506CD34F-0AD1-4421-9F98-BA19DC4E0253</string>
                         <key>SSID_STR</key>
                         <string>[% provisioner.ssid %]</string>
+                        [% IF provisioner.eap_type.undefined %]
+                        <key>Password</key>
+                        <string>[% passcode %]</string>
+                        [% END %]
+                </dict>
+                [% IF provisioner.eap_type.defined and provisioner.eap_type == "13" %]
+                <dict>
+                        <key>PayloadCertificateFileName</key>
+                        <string>[% certcn %]</string>
+                        <key>PayloadContent</key>
+                        [% IF for_windows %]
+                        <string>[% certdata %]</string>
+                        [% ELSE %]
+                        <data>[% certdata %]</data>
+                        [% END %]
+                        [% IF for_windows %]
+                        <key>PayloadDescription</key>
+                        <string>Provides device authentication (certificate or identity).[% fingerprint %]</string>
```

Comment:
Why a special case to put the fingerprint in for the Windows agent ? I hope you're not extracting the Fingerprint from this field in the Windows agent. If not then remove this